meson.build

```diff
@@ -1496,6 +1487,21 @@ sys.stdout.write(swift_ver)
     message('Detected Swift version: ' + swift_ver)
 endif
 
+macos_10_11_features = get_option('macos-10-11-features').require(
+    macos_sdk_version.version_compare('>=10.11'),
+    error_message: 'a suitable macos sdk version could not be found!',
+)
+
+macos_10_12_2_features = get_option('macos-10-12-2-features').require(
+    macos_sdk_version.version_compare('>=10.12.2'),
+    error_message: 'a suitable macos sdk version could not be found!',
+)
+
+macos_10_14_features = get_option('macos-10-14-features').require(
+    macos_sdk_version.version_compare('>=10.14'),
+    error_message: 'a suitable macos sdk version could not be found!',
+)
+
 swift = get_option('swift-build').require(
```

Comment:
you now have `swift` and `swift_build`

maybe make the checks depend on macos_sdk_version being set instead?

Reply:
Great, I'm bad with names. I definitely want to essentially pass `get_option('swift-build').require('darwin')` to those checks which is different from the `swift.allowed()` check.  Not sure what to call it. `swift_opt`?

Reply:
Simpler idea, would this work?

```py
macos_sdk_version_py = find_program(join_paths(source_root, 'TOOLS',
                                    'macos-sdk-version.py'), required: darwin)
# [...]
xcrun = find_program('xcrun', required: get_option('swift-build') and darwin)
# leave the rest as-is
```

Reply:
There's a slightly crappy error message if you try to enable `swift-build` on not macOS, but why would anyone do that anyways. Yeah this works.